### PR TITLE
Disable apply button in network tabs

### LINKED ISSUE #2724: Apply Button Always Enabled in Network
If you navigate to Network tab, you will see that the Apply button is always enabled, although user does not change anything, which is not the desired behvaiour. 

**Testflow:**
1. Login to Kura as admin/admin
2. Go to Network, without changing anything try to press Apply button
3. Apply button is clickable, and it looks like Kura saves something, but really doesn't.

**Expected behavior**
Apply button should be activated only when user makes some changes. 

**Screenshots**
Screenshot 2020-03-26 at 14 27 43

Target Environment (please complete the following information):
Board: Raspberry Pi 3
OS version: Linux raspberrypi 4.19.75-v7+ # 1270 SMP Tue Sep 24 18:45:11 BST 2019 armv7l GNU/Linux
Additional info: /

### PR DESCRIPTION
**Pull request by pierantoniomerlino**
This PR manages the activation/deactivation of the apply button on network tabs based on the dirty flag.

**Related Issue:** This PR fixes/closes #2724

### CODE REVIEW

**Comment by pierantoniomerlino** [comment]
@LeoNerdoG could you please check if this PR fixes the #2724?

**Comment by LeoNerdoG** [comment]
Hey @pierantoniomerlino,
I have checekd this branch and found several issues. 
**Testflow No. 1:**
1. In Networking tab select one of the networsk (e.g. ETH0)
2. Change one of the paramters (e.g. Status from Enabled for Wlan to Enabled for Lan)
3. Press "Refresh" button - observe the Apply button which is now disabled
4. Change views from eth0 to wlan0 and then come back - observe how changes have not been saved (which is OK, becasue we only pressed Refresh and not Apply)

**Testflow No. 2:**
1. In Networking, click on one of the free entry fields 
2. Apply button gets activated every time the focus is changed, although nothing has been changed

**Testflow No. 3:**
1. Go to Networking -> Wireless -> Channel list
2. Select different channel than default "11" - Apply button does not get activated

**Testflow No. 4:**
1. 1. Go to Networking -> Wireless -> "Ignore Broadcast SSID" list
2. Change setting from True to False - Apply button does nto get activated

Can you please check these scenarios? If I stumble upon something else, I will report.

**Comment by pierantoniomerlino** [comment]
@LeoNerdoG Thank you for the feedback!
**Testflow No. 1:** I think that is right, isn't it?

**Testflow No. 2:** I don't understand the test here: if I click on a field and I don't change it, the apply button remains disabled. Do you have more details on the procedure?

**Testflow No. 3:** and **Testflow No. 4:** Yes! There is a bug!

**Comment by LeoNerdoG** [comment]
Hey,
regarding No. 1: I think that this is an issue, because if user presses Refresh after changing something, Apply should be active? otherwie the "Refresh" button should be renamed to "Reset"

No. 2: Yeah, the problem here is that Apply gets activated just if you click on the field and change nothing (so ust focus is changed). This is not a critical error, but it could be a good improvement.

**Comment by pierantoniomerlino** [comment]
No. 1: yes, the "Refresh"  button indeed is used also for resetting the tabs. I'll leave it as is, for now.

No. 2: I admit that I can't see it. Have you a specific testflow?

**Comment by LeoNerdoG** [comment]
Hey @pierantoniomerlino,
regardin No. 2: if you just click on one of the free entry fields and afterwares outside, Apply button gets activated althoug nothing is changed - I think this should be fixed (so only if user cctually changes something the Apply button activates).

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java [hunk 1]
```diff
@@ -120,8 +121,8 @@
     private final GwtNetworkServiceAsync gwtNetworkService = GWT.create(GwtNetworkService.class);
     private final GwtDeviceServiceAsync gwtDeviceService = GWT.create(GwtDeviceService.class);
 
-    private static final String REGEX_PASSWORD_WPA = "^[ -~]{8,63}$";
-    private static final String REGEX_PASSWORD_WEP = "^(?:[\\x00-\\x7F]{5}|[\\x00-\\x7F]{13}|[a-fA-F0-9]{10}|[a-fA-F0-9]{26})$";
+    private static final String REGEX_PASS_WPA = "^[ -~]{8,63}$";
+    private static final String REGEX_PASS_WEP = "^(?:[\\x00-\\x7F]{5}|[\\x00-\\x7F]{13}|[a-fA-F0-9]{10}|[a-fA-F0-9]{26})$";
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:125]: Why do we need to change those names?

Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:125]: Sonar complains about the use of a constant with the name PASSWORD...

Comment (MMaiero) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:125]: Oh, ok.

**Comment by LeoNerdoG** [comment]
Hey @pierantoniomerlino,
Please let me know when this PR will be rady for testing.

**Comment by MMaiero** [comment]
Is this PR ready for a final review?

**Comment by pierantoniomerlino** [comment]
@MMaiero No :-(

**Comment by pierantoniomerlino** [comment]
@MMaiero @LeoNerdoG could you please take a look at this (endless) PR?

**Comment by LeoNerdoG** [comment]
@pierantoniomerlino I will check this right away.

**Comment by LeoNerdoG** [comment]
Hey @pierantoniomerlino,
I am confirming that this one is finally OK! 😄 
please go ahead an merge it!

**Review by MMaiero — APPROVED** [review]

**Pull request merged**